Sming/SmingCore/Network/Http/HttpRequest.h

```diff
@@ -30,82 +31,157 @@ typedef Delegate<int(HttpConnection& client, HttpResponse& response)> RequestHea
+	 */
+	HttpRequest* setPostParameters(const HttpParams& params)
+	{
+		postParams = params;
+		return this;
+	}
 
-	HttpRequest* setPostParameters(const HttpParams& params);
-	HttpRequest* setPostParameter(const String& name, const String& value);
+	HttpRequest* setPostParameter(const String& name, const String& value)
+	{
+		postParams[name] = value;
+		return this;
+	}
 
 	/**
 	 * @brief Sets a file to be sent
-	 * @param const String& name the name of the element in the form
+	 * @param const String& formElementName the name of the element in the form
 	 * @param FileStream* stream - pointer to the file stream
 	 *
 	 * @return HttpRequest*
 	 */
-	HttpRequest* setFile(const String& name, FileStream* stream);
+	HttpRequest* setFile(const String& formElementName, FileStream* stream)
+	{
+		if(stream) {
+			files[formElementName] = stream;
+		}
+		return this;
+	}
 
 #ifdef ENABLE_HTTP_REQUEST_AUTH
 	// Authentication adapters set here
-	HttpRequest* setAuth(AuthAdapter* adapter);
+	HttpRequest* setAuth(AuthAdapter* adapter)
+	{
+		adapter->setRequest(this);
+		auth = adapter;
+		return this;
+	}
 #endif
 
-	String getHeader(const String& name);
+	const String& getHeader(const String& name)
+	{
+		return static_cast<const HttpHeaders&>(headers)[name];
+	}
 
-	String getPostParameter(const String& name);
+	const String& getPostParameter(const String& name)
+	{
+		return static_cast<const HttpParams&>(postParams)[name];
+	}
 
-	__forceinline String getPath()
+	/* @deprecated  use uri methods */
+	String getPath()
 	{
 		return uri.Path;
 	}
 
-	String getQueryParameter(const String& parameterName, const String& defaultValue = "");
+	/* @deprecated  use uri methods */
+	String getQueryParameter(const String& parameterName, const String& defaultValue = nullptr);
 
 	/**
 	 * @brief Returns content from the body stream as string.
 	 * @retval String
 	 *
 	 * @note This method consumes the stream and it will work only with text data.
 	 * 		 If you have binary data in the stream use getBodyStream instead.
+	 *
+	 * @note Allocation of String doubles amount of memory required, so use with care.
 	 */
 	String getBody();
 
 	/**
-	 * @brief Returns pointer to the current body stream
+	 * @brief Return the current body stream and pass ownership to the caller
 	 * @retval ReadWriteStream*
+	 * @note may return null
 	 */
 	ReadWriteStream* getBodyStream();
 
-	HttpRequest* setBody(const String& body);
+	HttpRequest* setBody(const String& body)
+	{
+		setBody((uint8_t*)body.c_str(), body.length());
+		return this;
+	}
+
 	HttpRequest* setBody(ReadWriteStream* stream);
+
 	HttpRequest* setBody(uint8_t* rawData, size_t length);
 
 	/**
 	 * @brief Instead of storing the response body we can set a stream that will take care to process it
 	 * @param ReadWriteStream *stream
 	 *
 	 * @retval HttpRequest*
+	 *
+	 * @note The response to this message will be stored in the user-provided stream.
```

Comment:
`The response to this message`. Did you mean `The response to this request` ?

Reply:
Yes! Well spotted.